src/Types.hs

```diff
@@ -0,0 +1,75 @@
+module Types where
+
+import Cardano.Prelude
+
+--import qualified Katip as Katip
+import qualified System.Metrics as Ekg
+
+-- | Cardano configuration
+data CardanoConfiguration = CardanoConfiguration
+    { scfgLogPath   :: !FilePath
+    , scfgDBPath    :: !FilePath
+    , scfgSomeParam :: !Int
+    }
+
+-- | The common runtime environment for all features in the server.
+-- All features have access to this environment.
+data CardanoEnvironment = CardanoEnvironment
+    { serverLogEnv      :: Text --Katip.LogEnv
+    , serverEkgStore    :: Ekg.Store
+     -- ...
+    }
+
+-- | Initalise 'ServerEnv'
+initializeCardanoEnvironment :: IO CardanoEnvironment
+initializeCardanoEnvironment = do
+  --  logenv   <- Katip.initLogEnv (...) (...)
+    ekgStore <- Ekg.newStore
+    return CardanoEnvironment {serverLogEnv = "To implement", serverEkgStore = ekgStore}
+
+loadCardanoConfiguration :: IO CardanoConfiguration
+loadCardanoConfiguration = do
+    pure $ CardanoConfiguration mempty mempty 0
+
+
+-- | The feature types we can have in the project.
+data FeatureType
+    = LoggingMonitoringFeature
+    | NetworkingFeature
+    | BlockchainFeature
+    | LedgerFeature
+    | WalletFeature
+    deriving (Eq, Show)
```

Comment:
Do we need this or is it just for information? If it's just for info I'd use a string instead. The framework should not generally have to know the complete list of features.

Reply:
Just type safe info. Yes, there is a tradeoff here, but we will have a point where we will have to define all the features so it might be a good idea to have them (type)checked.